libraries/shared/fetcher/state_diff_fetcher.go

```diff
@@ -0,0 +1,73 @@
+// VulcanizeDB
+// Copyright © 2019 Vulcanize
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package fetcher
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/statediff"
+
+	"github.com/vulcanize/vulcanizedb/pkg/eth/client"
+)
+
+// StateDiffFetcher is the state diff fetching interface
+type StateDiffFetcher interface {
```

Comment:
I'm wondering with the addition of this fetcher, if we want to rethink the naming of the `GethRpcStorageFetcher` and/or the `StorageFetcherInterface`. For me there is some confusion because the `GethRpcStorageFetcher` is connecting to the geth subscription, which is technically emitting statediffs as well.  But maybe this is a specific implementation detail that the interface is doing a good job obscuring? 

And maybe `GethRpcStorageFetcher` can go away at some point even, so this distinction doesn't matter? 🤷‍♀ 

This isn't super relevant to this PR, but I'm curious about your thoughts about it.

Reply:
I'm not sure what the best way to sort out these interfaces is but seems like a good discussion to have. To me both the `GethRpcStorageFetcher` and the `CsvTailStorageFetcher` are more "streamers" than "fetchers" in that they are handling subscriptions and feeding data out to a channel. Another distinction is that `GethRpcStorageFetcher` and `CsvTailStorageFetcher` are returning `utils.StorageDiff`s whereas the  `StateDiffFetcher` is returning unprocessed `statediff.Payload`s. Also, the `StateDiffFetcher` is using the `statediff_stateDiffAt` http rpc endpoint whereas the `GethRpcStorageFetcher` is using the `StateDiffStreamer` which works with the `statediff_stream` ws rpc endpoint.